### PR TITLE
Feature/flag api

### LINKED ISSUE #1013: Flag inappropriate or defunct APIs
# User story
> As an Apinf administrator I want to flag inapproprite or defunct APIs so that it becomes possible to exclude or filter out these APIs

# Definition of done

## Collection
* [x] Collection exists for flags
  * [x] Contains flag reason: inappropriate, defunct
  * [x] Contains a text field for description or comment
  * [x] Contains reference field for user id (which admin flagged the API)
  * [x] foreign key to the API backend getting flagged
* [ ] if the api is unflagged by admin, the information gets updated to db

## UI component
* [x] UI component for flagging (opens dialog)
* [x] Modal dialog for adding the flag
* [ ] Modal dialog contains an option to unflag

## User permissions
 * [x] Only administrator can flag
* [x] Only administrator can view the UI component and the dialog
* [x] Only administrator can insert anything to the collection
* [ ] Only administrator can unflag

## Outcome
* [ ] Flagged API is not visible in catalog **(Update: decided to keep this open until the new version of api catalogue list is implemented)**

### PR DESCRIPTION
**Pull request by frenchbread**
Closes #1013 

# Additions
- UI button to flag API Backend
- ApiFlags collection, schema & permissions
- Flag Api backend form - modal
- i18n tags

### CODE REVIEW

**Comment by frenchbread** [comment]
@apinf/developers Please review.

**Comment by elnzv** [comment]
We need to allow admin
- update flag message (reason)
- unflag API

Example:
screencapture-localhost-3000-api-rygy2tp8agbw6geos-1465462453492

File: flags/collection/schema.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+ApiFlags.schema = new SimpleSchema({
+  reason: {
+    type: String,
+    allowedValues: [
+      'Inappropriate',
```
Comment (brylie) [inline flags/collection/schema.js]: We will need to internationalize these strings in the UI. The `allowedValues` texts can be in English, but the UI element text should change depending on the active language.

**Comment by frenchbread** [comment]
Remove `WIP`. Please review.

**Pull request merged**